proto/minder/v1/minder.proto

```diff
@@ -2770,12 +2770,22 @@ message Provider {
         (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
     ];
     // class is the name of the provider implementation, eg. 'github' or 'gh-app'.
-    string class = 7;
+    string class = 7 [
+        (buf.validate.field).string = {
+            pattern: "^[a-z][a-z0-9_-]*$",
+            max_len: 200,
+        },
+        (buf.validate.field).ignore = IGNORE_IF_DEFAULT_VALUE
+    ];
     // project is the project where the provider is.  This is ignored on input
     // in favor of the context field in CreateProviderRequest.
     string project = 2;
     // version is the version of the provider.
-    string version = 3;
+    string version = 3 [
+        (buf.validate.field).string = {
+            pattern: "^v\\d+(\\.\\d+)?(\\.\\d+)?(?:-[\\w\\.-]+)?$",
```

Comment:
It looks like our only current allowed value is `v1`.  Let's not permit dot-syntax until we have evidence that it's needed.

Reply:
Updated it to enforce `v1` 👍